File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int tmp;

bool cmp(vector<int> a, vector<int> b) {
    return a[tmp] < b[tmp];
}

vector<vector<int>> solution(vector<vector<int>> data, string ext, int val_ext, string sort_by) {
    vector<vector<int>> answer;
    
    if(ext == "code")
    tmp  = 0;
    else if (ext == "date")
    tmp = 1;
    else if (ext == "maximum")
    tmp = 2;
    else
    tmp = 3;
    
    for(auto i : data) {
        if(i[tmp] < val_ext)
            answer.push_back(i);
    }
    
    if(sort_by == "code")
    tmp  = 0;
    else if (sort_by == "date")
    tmp = 1;
    else if (sort_by == "maximum")
    tmp = 2;
    else
    tmp = 3;
    
    sort(answer.begin(), answer.end(), cmp);
    
    return answer;
}

File: 프로그래머스/lv1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;

class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        Map<String, Integer> map = new HashMap<>();
        map.put("code", 0);
        map.put("date", 1);
        map.put("maximum", 2);
        map.put("remain", 3);
        
        List<int[]> list = new ArrayList<>();
        
        for (int[] i : data) {
            if (i[map.get(ext)] < val_ext)
                list.add(i);
        }
        
        Collections.sort(list, (o1, o2) -> o1[map.get(sort_by)] - o2[map.get(sort_by)]);
        
        int[][] ans = new int[list.size()][];
        int idx = 0;
        for (int[] i : list)
            ans[idx++] = i;
        
        return ans;
    }
}